### DISCUSSION TITLE
Potential occasional crash / lock with high CPU use

### DISCUSSION THREAD

**Discussion by krisnoble** [music-assistant/support#1530] [Q&A]
Not really sure where to put this, searching discussions and issues hasn't turned up anything that seems related so figured I'd document it just in case it's useful in triaging or if anyone else has similar problems.

I'm seeing occasional crash / lock type behaviour where MA seems to get hung doing something and causes a sustained high CPU usage while simultaneously being unavailable for playback or UI.

In the graphs here you can see a simultaneous raised CPU temp, usage and power draw respectively, around the middle of the chart. The memory use doesn't seem to be affected. The increased activity here lasts about an hour (until I noticed and restarted the add-on) as opposed to the other momentary spikes. 

<img width="1007" alt="Screenshot 2023-09-15 at 14 20 00" src="https://github.com/music-assistant/hass-music-assistant/assets/872971/94a3ca0d-8461-414a-9efa-9e31a5b40d24">

In this case I'm running MA as an add-on in a Proxmox VM with 2 cores of an i5-2340m @ 2.4GHz and 4GB memory. Hard drive is a SATA SSD. I was able to restart the add-on from HA. Anecdotally before I moved to my current setup I did have a crash on my Raspberry Pi 4 running Docker, which locked the whole device although obviously difficult to pin that down conclusively to MA. I suspect it's the same though hence flagging it now.

I have local file and radio browser providers, and Slimproto players. I didn't see anything significant in the logs around the time of the issue. I wasn't playing synced audio at the time but was playing different audio on two players. I didn't catch it straight away because I'd muted the room I was in for a phone call 🤦🏻‍♂️

Think that covers everything relevant but let me know if I can dig up any more details. If it happens again I'll try to capture more data, logs etc.

**Comment by OzGav** [comment]
Thanks. You will need to try and recreate it as we don’t have any other reports of this type at the moment.

**Reply by krisnoble** [reply]
Yeah no sweat, just adding the data point really. Hopefully it ends up being nothing anyway 🤞🏻

**Comment by krisnoble** [comment]
Just an update, I've submitted an issue #1566 that may be related - can't say for certain that it was the same problem but plausible and repeatable at my end at least
